contracts/contracts/RestrictedCalls.sol

```diff
@@ -30,12 +53,29 @@ contract RestrictedCalls is Ownable {
     ///
     /// Example usage::
     ///
-    ///     restricted(block.chainid, msg.sender)
+    ///     restricted(block.chainid)   // expecting calls from the same chain
+    ///     restricted(otherChainId)    // expecting calls from another chain
     ///
-    modifier restricted(uint256 chainId, address caller) {
-        bytes32 key = keccak256(abi.encodePacked(chainId, caller));
+    modifier restricted(uint256 callerChainId) {
+        address caller = callers[callerChainId][0];
 
-        require(callers[key], "RestrictedCalls: unknown caller");
+        if (callerChainId == block.chainid) {
+            require(msg.sender == caller, "RestrictedCalls: unknown caller");
```

Comment:
can'T we unify the revert message with the one below? "RestrictedCalls: call disallowed"

Reply:
Good point. Done.